wrapper/src/main/java/software/amazon/jdbc/plugin/failover/FailoverConnectionPlugin.java

```diff
@@ -213,6 +218,19 @@ public <T, E extends Exception> T execute(
       final JdbcCallable<T, E> jdbcMethodFunc,
       final Object[] jdbcMethodArgs)
       throws E {
+
+    try {
+      if (this.enableFailoverSetting
+          && !this.canDirectExecute(methodName)
+          && !this.closedExplicitly.get()
+          && this.pluginService.getCurrentConnection() != null
+          && this.pluginService.getCurrentConnection().isClosed()) {
+        this.pickNewConnection();
```

Comment:
Should we log here seeing as this is a bit of a special scenario? Also, should we add a configuration parameter to enable/disable this? Not sure if some customers might prefer to error out if they end up executing against a closed connection.

Reply:
> I realize that hostListRefreshThresholdTimeNano may change since it is static but I don't understand why we set it to 0 in the compareAndSet call if it hasn't changed in the last few lines of code. Why does that mean we should clear the value?

I added comments to the code. I hope it adds clarity. The code sets a time value to hostListRefreshThresholdTimeNano when it's necessary to keep refreshing host list. When the specified time is reached, we reset hostListRefreshThresholdTimeNano to zero and deactivates further host list refreshes.

Reply:
> Should we log here seeing as this is a bit of a special scenario? Also, should we add a configuration parameter to enable/disable this? Not sure if some customers might prefer to error out if they end up executing against a closed connection.

Failover code has necessary logging so I believe it provides enough details for troubleshooting. I don't think that user has a choice here. Without failover in this case, user (most probably) gets an error executing JDBC call on closed connection. With this code, user may get a chance to failover and proceed with JDBC call successfully.

Reply:
Right, I was thinking that for customer logs it might be helpful to know that failover was kicked off because of executing against a closed connection which may be different than executing against a network-failed connection. Up to you though.